plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/AbstractResourceMojo.java

```diff
@@ -92,30 +93,54 @@ protected void writeResources(KubernetesList resources, ResourceClassifier class
         }
     }
 
-    public static File writeResourcesIndividualAndComposite(KubernetesList resources, File resourceFileBase, ResourceFileType resourceFileType, Logger log) throws MojoExecutionException {
+    public static File writeResourcesIndividualAndComposite(KubernetesList resources, File resourceFileBase, ResourceFileType resourceFileType, Logger log, Boolean generateRoute) throws MojoExecutionException {
+
+        List<HasMetadata> oldItemList = resources.getItems();
+
+        List<HasMetadata> newItemList = new ArrayList<>();
+
+        if(!generateRoute) {
+
+            for (HasMetadata item : resources.getItems()) {
+                if (item.getKind().equalsIgnoreCase("Route")) {
+                    continue;
+                }
+                newItemList.add(item);
+            }
+
+            resources.setItems(newItemList);
+
+        }
+
         Object entity = resources;
+
         // if the list contains a single Template lets unwrap it
         Template template = getSingletonTemplate(resources);
         if (template != null) {
             entity = template;
         }
+
         File file = writeResource(resourceFileBase, entity, resourceFileType);
 
         // write separate files, one for each resource item
-        writeIndividualResources(resources, resourceFileBase, resourceFileType, log);
+        resources.setItems(oldItemList);
+        writeIndividualResources(resources, resourceFileBase, resourceFileType, log, generateRoute);
         return file;
     }
 
-    private static void writeIndividualResources(KubernetesList resources, File targetDir, ResourceFileType resourceFileType, Logger log) throws MojoExecutionException {
+    private static void writeIndividualResources(KubernetesList resources, File targetDir, ResourceFileType resourceFileType, Logger log, Boolean generateRoute) throws MojoExecutionException {
         for (HasMetadata item : resources.getItems()) {
             String name = KubernetesHelper.getName(item);
             if (Strings.isNullOrBlank(name)) {
                 log.error("No name for generated item %s", item);
                 continue;
             }
             String itemFile = KubernetesResourceUtil.getNameWithSuffix(name, item.getKind());
-            File itemTarget = new File(targetDir, itemFile);
-            writeResource(itemTarget, item, resourceFileType);
+
+            if (!(item.getKind().equalsIgnoreCase("Route") && generateRoute.equals(false))){
```

Comment:
You can have a cleaner check via `item instanceof Route`

Reply:
I think this will not make much difference and here items are the types of HasMetadata which are indirectly every OpenShift resource. Accidentally, we may not end up deleting any other resource, i used kind value which will be unique.

Reply:
I think we can use `instanceof` for both Kubernetes and Openshift resources.

Reply:
Problem is not we can use instance of with both Kubernetes and Openshift. My case is only related to Openshift as Route is only available in OpenShift. We may not end up deleting some other resource I am using very more accurate defined condition where Kind is equal to Route